Controller/OneClick/PlaceOrder.php

```diff
@@ -323,6 +329,20 @@ public function execute()
             ];
             $orderNotes = array_merge($orderNotes, $customerEmailNotes);
         }
+        if($giftCardDiscountAmount > 0)
+        {
+            $totalAmount = $totalAmount - $giftCardDiscountAmount;
+            $giftCardDiscount = [
+                'gift_card_discount'=> $giftCardDiscountAmount,
+            ];
+            $orderNotes = array_merge($orderNotes, $giftCardDiscount);
+        }
+        $promotions = [
+            'type' => 'gift_card',
+            'code' => 'gift_card_number',
+            'value' => $giftCardDiscountAmount,
+            'description' => 'applied',
+        ];
```

Comment:
When there's no Gift Card Discount applied, I believe the `$promotions` should be empty
```suggestion
        $promotions = [];
        
        if($giftCardDiscountAmount > 0)
        {
            $totalAmount = $totalAmount - $giftCardDiscountAmount;
            $giftCardDiscount = [
                'gift_card_discount'=> $giftCardDiscountAmount,
            ];
            $orderNotes = array_merge($orderNotes, $giftCardDiscount);
            
            $promotions = [
                'type' => 'gift_card',
                'code' => 'gift_card_number',
                'value' => $giftCardDiscountAmount,
                'description' => 'applied',
            ];
        }
```